src/components/TradeDetails/TradeDetails.tsx

```diff
@@ -4,40 +4,48 @@ import { TokenInfo } from '@airswap/typescript';
 import { format } from '@greypixel_/nicenumbers';
 import { BigNumber } from 'bignumber.js';
 
+import { CollectionToken } from '../../entities/CollectionToken/CollectionToken';
+
 import './TradeDetails.scss';
 
 interface TradeDetailsProps {
   amount?: BigNumber;
   amountSubtext?: string;
   title: string;
-  tokenInfo: TokenInfo;
+  token: TokenInfo | CollectionToken;
   className?: string;
 }
 
 const TradeDetails: FC<TradeDetailsProps> = ({
   amount,
   amountSubtext,
   title,
-  tokenInfo,
+  token,
   className = '',
 }) => {
+  const logoURI = 'logoURI' in token && token.logoURI;
+  const image = 'image' in token && token.image;
+  const symbol = 'symbol' in token && token.symbol;
+  const name = 'name' in token && token.name;
+  const id = 'id' in token && token.id;
+
   const readableAmount = useMemo(() => {
-    if (!amount) {
+    if (!amount || !('decimals' in token)) {
```

Comment:
I feel like you may as well have defined the decimals part with the consts just above.. But :shrug: